beaker_kernel/lib/agent.py

```diff
@@ -40,6 +40,11 @@ def __init__(
             thought_handler=context.beaker_kernel.handle_thoughts,
             **kwargs
         )
+        if not env_enabled("ENABLE_USER_PROMPT"):
```

Comment:
By continuing to gate these behind an environment variable, it's not possible, or at least straightforward, to turn these on/off on a per-context basis.

Reply:
Just to double check, would the idea here be that all the contexts in askem-beaker have these lines instead?

Reply:
The idea is that there would be a default and if different contexts wanted to deviate from the default there would be a way to do so.

The __init__ function allows **kwargs for exactly this sort of expansion. It would be a great method to allow the ability to override, but it could be a class attribute too. Maybe a combination of the two?

But no, the idea is not to have all the two lines in each context. For today, all the code should be in this file.

Reply:
Inspired by your suggestion [here](https://jataware.slack.com/archives/C03J0RSKFK8/p1715262724438199?thread_ts=1715196514.594459&cid=C03J0RSKFK8), I implemented tool toggling via class attributes. Lmk what you think!

![sample](https://github.com/jataware/beaker-kernel/assets/14170067/3a53da87-3d78-4863-874f-ee771c8b0fff)

Reply:
this conversation has been moved out into the primary thread. closing this for now